src/goto-instrument/code_contracts.cpp

```diff
@@ -127,9 +137,11 @@ void code_contractst::check_apply_invariants(
   build_havoc_code(loop_head, modifies, havoc_code);
 
   // assume the invariant
-  code_assumet assumption{invariant};
-  assumption.add_source_location() = loop_head->source_location;
-  converter.goto_convert(assumption, havoc_code, mode);
+  {
+    code_assumet assumption{invariant_expr()};
+    assumption.add_source_location() = loop_head->source_location;
+    converter.goto_convert(assumption, havoc_code, mode);
+  }
```

Comment:
Why are you adding `{...}`?

Reply:
For scoping variables.

The invariant gets expanded to multiple assertions and assumptions. I scope these, for example the `assumption` object in this case, so they don't pollute the scope of the overall function. They are immediately compiled to goto instructions and added to `havoc_code`.